sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample14_ReadEventHubsEventsUsingAzureEventSourceListener.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Azure.Messaging.EventHubs.Samples
+{
+    /// <summary>
+    /// An example of catching all Event Hubs events using AzureEventSourceListener and send them to Application Insights.
+    /// </summary>
+    public class Sample14_ReadEventHubsEventsUsingAzureEventSourceListener : IEventHubsApplicationInsightsSample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name => nameof(Sample14_ReadEventHubsEventsUsingAzureEventSourceListener);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description => "An example of catching all Event Hubs events using AzureEventSourceListener and send them to Application Insights.";
+
+        public async Task RunAsync(string connectionString,
+                      string eventHubName,
+                      string instrumentationKey)
+        {
+            // Setup a listener to monitor logged events.
+            var listener = new AzureEventSourceListener((args, message) =>
+            {
+                // Add condition to manipulate only events related to EventHubs.
+                if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+                {
+                    TelemetryConfiguration configuration = TelemetryConfiguration.CreateDefault();
+                    configuration.InstrumentationKey = instrumentationKey;
+                    var telemetryClient = new TelemetryClient(configuration);
+                    var properties = args.PayloadNames.Zip(args.Payload, (k, v) => new { Key = k, Value = v })
+                      .ToDictionary(x => x.Key, x => x.Value.ToString());
+                    var level = GetSeverityLevelFromEventLevel(args.Level);
+                    var traceMessage = string.Format(args.Message, args.Payload.ToArray());
+                    telemetryClient.TrackTrace($"Sample_14 AzureEventSourceListener | {traceMessage}", level, properties);
+                }
+            }, EventLevel.LogAlways);
+
+            // Creating a producer client using its default set of options.
+            // We will publish three small batch events.
+
+            await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
```

Comment:
We probably want to move away from the async disposable;  there has been some confusion about whether the client is safe to cache and use for the life of an application.  This may be a good opportunity to create the producer and telemetry client together and then call `CloseAsync` when the application run is ending.